autorest/adal/cli.go

```diff
@@ -47,13 +50,42 @@ func AzureCLIProfilePath() (string, error) {
 }
 
 // ToToken converts an AzureCLIToken to a Token
-func (t AzureCLIToken) ToToken() Token {
-	return Token{
+func (t AzureCLIToken) ToToken() (*Token, error) {
+	tokenExpirationDate, err := ParseAzureCLIExpirationDate(t.ExpiresOn)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing Token Expiration Date %q: %+v", t.ExpiresOn, err)
+	}
+
+	difference := tokenExpirationDate.Sub(expirationBase)
+	seconds := difference.Seconds()
+
+	token := Token{
 		AccessToken:  t.AccessToken,
 		Type:         t.TokenType,
-		ExpiresIn:    strconv.Itoa(t.ExpiresIn),
-		ExpiresOn:    t.ExpiresOn,
+		ExpiresIn:    "3600",
+		ExpiresOn:    strconv.Itoa(int(seconds)),
```

Comment:
[nit] I think this would be more readable as `difference.Seconds()` and skipping the declaration of `seconds`

Reply:
Updated :)